### DISCUSSION TITLE
Any way to import hand-written recipes?

### DISCUSSION THREAD

**Discussion by mlcampbe** [mealie-recipes/mealie#2102] [General]
### First Check

- [X] This is not a feature request
- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.
- [X] I have checked for existing issues that have been resolved in v1-beta

### What is the issue you are experiencing?

I am a new mealie user and trying to find the best way to import our hand-written receipes. Is there a way to take a picture or create a pdf of a hand-written recipe and get it to import successfully?

### Deployment

Docker (Linux)

### Mealie Version

0.5.6

**Comment by Wetzel402** [comment]
I would expect the best way to do something like this would be with AI image analysis.  Similar to how business card scanning apps work.

**Comment by realthore** [comment]
Late reply but I had the same issue and found ChatGPT 4o pretty useful for that. Got some insanely bad handwriting from my granny and it was able to read it. Without GPT+ subscription you're pretty limited but I have a chat with a prompt like "I'll give you multiple pictures of handwritten recipes. List all ingredients and the individual steps. Also add total time, prep time and cook/bake time (or estimate if not available). Write a short description as well"

**Reply by Kuchenpirat** [reply]
There is a PR open for that (See #3974).
It does rely on having an OpenAI Api key, so there will be costs when importing via this method. But they should not be that high. 

Otherwise you can try to get chatgpt to stick to recipe schema and extract that info in a way that you can use with the mealie api to create the recipes this way.

**Reply by realthore** [reply]
Looks promising. Do you know how to deploy mealie next in a docker environment? ghcr.io/mealie-recipes/mealie:next doesn't work for me

**Reply by michael-genson** [reply]
You're looking for `ghcr.io/mealie-recipes/mealie:nightly`

**Comment by Thunderstrike116** [comment]
Personally, I think the best way to import hand-written recipes is to use some kind of voice typing software (like google docs or microsoft word) and read aloud your recipes to it. Then you can make any corrections required and just copy and paste everything in Mealie.
I haven't tested ChatGPT or other AI tools, but my method is absolutely free and maybe more reliable .

**Comment by michael-genson** [comment]
The newest nightly has the ability to import via image using OpenAI: https://github.com/mealie-recipes/mealie/pull/3974; it can also optionally translate the recipe for you (with varying success depending on which model you use)
